### DISCUSSION TITLE
[@types/react] Return type of forwardRef is missing the "render" property

### DISCUSSION THREAD

**Discussion by EduardoSimon** [DefinitelyTyped/DefinitelyTyped#68505] [Issues with a @types package]
Hi there, I'm reopening issue #50783 as a discussion. 

The link in the original issue pointing to the type is wrong, here's the [permalink](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9bbff85ffe3d5f31394e324e3cb4cb57a4d37599/types/react/index.d.ts#L1474)

In react the forwardRef return object is [here](https://github.com/facebook/react/blob/9198a5cec0936a21a5ba194a22fcbac03eba5d1d/packages/react/src/ReactForwardRef.js#L48). Would you accept a contribution that adds the render prop to the `ForwardRefExoticComponent` type?

Regards,
Eduardo

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
The returned value from `forwardRef` is supposed to be an opaque object whose internal properties are only known to React. User space code is not supposed to know about the `render` property since there's no legit use case to call that method.

Due to TypeScript's limitations we have to make it appear like a function but conceptually, its properties are not supposed to be known nor used.

**Reply by EduardoSimon** [reply]
Thanks for the quick response! I didn't come across this issue until I tried to mock a component that uses forward ref. I cant put together an example to illustrate it if necessary.
